### PR TITLE
Many updates for v2, see PR description

### PR DESCRIPTION
**Pull request by IlCallo**
## Linting
I tinkered with ESLint config to only apply TS linting to TS files, avoiding false positives for JS files

## File generation
I moved the boot file to be generated into actual `boot` folder.
I guess you put it into the "apollo" folder to automatically register it via the AE index file, but you can do so only if you have full control over the file, eg. if you don't expose it.
If you do expose it, it may need to be applied in a different order depending on what the dev does into it.
Otherwise you're just going against the boot files pattern and adding mental burden for the user to the process.

I also removed `extension` containing folder as we never used this convention on other official AEs AFAIK, what's the rationale behind it?

I also set the typescript prompts to false by default, as only 25% of Quasar userbase uses TS right now. We can change this depending on next years community surveys.

## Typings
I fixed some typings here and there, eg. you don't need `BootFileParams` if you use `boot` wrapper.

Old `extensions/apollo/index.d.ts ` isn't needed anymore, as forcing TS to bring in `quasar` typings into `tsconfig.json` automatically imports all needed types.

I had to disable lib check due to chain-webpack types conflicting with webpack5 ones. It's strange as we don't get this problem on generated projects, honestly couldn't understand which is the problem and why it happens there.

##  Deps
I updated all deps to latest version.
I removed "react" dep as importing stuff from '@apollo/client/core' instead of '@apollo/client' removes the need for it.
I removed "typescript" dep as '@quasar/app' already provides it.

## Repo mangement
I added `yarn format` to apply Prettier formatting everywhere.

I added `yarn build` to automatically clean the lib folder, generate the updated JS files versions and format them with Prettier. In this way we lose the newlines in generated JS files (until TS guys figure out a way to allow it) but a lot of repo management time and mental burden is removed.

I added `yarn deploy` scripts to automatize the publishing process, in particular to avoid forgetting to generate JS files and using the wrong npm tag.

## Installation steps
I used [`app.provide`](https://v3.vuejs.org/api/application-api.html#provide) into the boot file, this avoids the need to update `App.vue` (which was needed when using Vue2 as there wasn't a global provide API).

## Left out / WIP
I also want to add tooling support for https://www.apollographql.com/docs/devtools/apollo-config/ and automatically scaffold an example one, as well as injecting `apollographql.vscode-apollo` into vscode suggested extensions files if present

Subscriptions via WebSocket need a particular setup, it could be asked at installation time or a guide could be integrated into the README

---

Let me know what do you think of the changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,12 +13,12 @@ It uses [Apollo Client](https://www.apollographql.com) and [Vue Apollo](https://
...
+**Note:** You need to use the `@next` tag until the final version of v2 of Quasar is released. At that point, we'll be moving v2 of the Apollo AE to the "latest" version and you will be able to install it without the `@next` tag. At that point, in order to install the older versions of the Apollo AE, you will need to add the version tag. This version will also stay in beta until Vue-Apollo is final (out of alpha or beta).
 
 Quasar CLI will retrieve the extension from NPM
 ([@quasar/quasar-app-extension-apollo](https://www.npmjs.com/package/@quasar/quasar-app-extension-apollo))
 
-The extension will add a directory `src/extensions/apollo`.
+The extension will add a configuration file into `src/apollo` and a boot file, register the latter into `quasar.conf.js > boot`.
```
Comment (smolinari) [inline README.md:21]: Could you please update this sentence to the following: 

The extension will add a configuration file into `src/apollo`, also a boot file, and it will register the latter into `quasar.conf.js > boot`.

Thanks. 

Scott

File: package.json [hunk 2]
```diff
@@ -2,7 +2,11 @@
...
   "description": "A Quasar app extension to add GraphQL support using Apollo Client.",
-  "keywords": ["Quasar", "GraphQL", "Apollo"],
+  "keywords": [
+    "quasar",
+    "graphql",
+    "apollo"
+  ],
   "contributors": [
     "Scott Molinari <scott.molinari@adduco.de>",
     "Ejez <ejez@protonmail.ch>"
```
Comment (smolinari) [inline package.json:12]: Could you add your name here too please @IlCallo. :)

Scott

**Review by smolinari — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (IlCallo) [reply README.md:21]: I removed the auto-registering of the boot file. If you allow the dev to change the boot file code at will, you must allow him to choose the order of execution too, so a manual registration is needed.

Maybe I should reword it to make it clearer the dev needs to manually register it?

File: package.json (hunk 2 above)
Comment (IlCallo) [reply package.json:12]: Will do, but only because you asked 🤣

**Comment by IlCallo** [comment]
I removed the need to update App.vue file, check out the updated PR description and latest commits

File: README.md (hunk 1 above)
Comment (smolinari) [reply README.md]: Sure. I was just concerned with the grammar mainly though. :)

Scott

Comment (IlCallo) [reply README.md]: Is it ok as I put it now?

**Comment by smolinari** [comment]
@IlCallo - merged. I'll wait for your further improvements before publishing the next release. Thanks a lot for your help!!! :)

Scott
